### PR TITLE
substrate: return Option in all storage related RPC methods

### LINKED ISSUE #508: getStorage RPC should return None rather than Err if value non-extant

### PR DESCRIPTION
**Pull request by andresilva**
Fix #508.

### CODE REVIEW

File: substrate/client/src/client.rs [hunk 1]
```diff
@@ -214,16 +214,15 @@ impl<B, E, Block> Client<B, E, Block> where
...
-			.ok_or_else(|| error::ErrorKind::NoValueForKey(key.0.clone()))?
-			.to_vec()))
+			.map(StorageData))
 	}
 
 	/// Get the code at a given block.
-	pub fn code_at(&self, id: &BlockId<Block>) -> error::Result<Vec<u8>> {
-		self.storage(id, &StorageKey(b":code".to_vec())).map(|data| data.0)
+	pub fn code_at(&self, id: &BlockId<Block>) -> error::Result<Option<Vec<u8>>> {
+		Ok(self.storage(id, &StorageKey(b":code".to_vec()))?.map(|data| data.0))
```
Comment (andresilva) [inline substrate/client/src/client.rs]: Can we expect this key to always exist? (even if set to empty).

Comment (pepyakin) [reply substrate/client/src/client.rs]: Yeah, definitely. Under this key the code of the runtime is stored. 

For example, [here is a genesis storage](https://raw.githubusercontent.com/paritytech/polkadot/master/polkadot/service/res/krummelanke.json) for krummelanke (Key `0x3a636f6465` is actually stands for ":code")

Comment (andresilva) [reply substrate/client/src/client.rs]: Should I call `expect` here and return `Result<Vec<u8>>` instead?

Comment (pepyakin) [reply substrate/client/src/client.rs]: I would answer "yes". Personally, I don't see any legal use-cases for it to be `None`

**Pull request merged**